pkg/reconciler/scheduling/placement/placement_controller.go

```diff
@@ -219,91 +165,69 @@ type controller struct {
 	namespaceLister  corelisters.NamespaceLister
 	namespaceIndexer cache.Indexer
 
-	apiBindignLister  apislisters.APIBindingLister
-	apiBindingIndexer cache.Indexer
-
 	locationLister  schedulinglisters.LocationLister
 	locationIndexer cache.Indexer
 
-	workloadClusterLister  workloadlisters.WorkloadClusterLister
-	workloadClusterIndexer cache.Indexer
+	placmentLister   schedulinglisters.PlacementLister
+	placementIndexer cache.Indexer
 }
 
-// enqueueLocationDomain enqueues all namespaces.
-func (c *controller) enqueueAPIBinding(obj interface{}) {
-	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-	clusterName, bindingName := clusters.SplitClusterAwareKey(key)
-
-	namespaces, err := c.namespaceIndexer.ByIndex(byWorkspace, clusterName.String())
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	for _, obj := range namespaces {
-		ns := obj.(*corev1.Namespace)
-		klog.Infof("Queueing namespace %s|%s because APIBinding %q changed", clusterName.String(), ns.Name, bindingName)
-		key := clusters.ToClusterAwareKey(logicalcluster.From(ns), ns.Name)
-		c.queue.Add(key)
-	}
-}
-
-func (c *controller) enqueueNamespace(obj interface{}) {
+func (c *controller) enqueuePlacement(obj interface{}) {
 	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 	clusterName, name := clusters.SplitClusterAwareKey(key)
 
-	klog.Infof("Queueing Namespace %s|%s", clusterName.String(), name)
+	klog.Infof("Queueing Placement %s|%s", clusterName.String(), name)
 	c.queue.Add(key)
 }
 
-func (c *controller) enqueueLocation(obj interface{}) {
+// enqueueNamespace enqueues all placements for the namespace
+// the namespace should have an annotation "scheduling.kcp.dev/placement" with the value as
+// {"placement1": state, "placement2": state}
+func (c *controller) enqueueNamespace(obj interface{}) {
 	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
-	clusterName, name := clusters.SplitClusterAwareKey(key)
+	clusterName, ns := clusters.SplitClusterAwareKey(key)
 
-	nss, err := c.namespaceIndexer.ByIndex(byNegotiationWorkspace, clusterName.String())
+	placements, err := c.placementIndexer.ByIndex(byWorkspace, clusterName.String())
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
-	for _, obj := range nss {
-		ns := obj.(*corev1.Namespace)
-		key := clusters.ToClusterAwareKey(logicalcluster.From(ns), ns.Name)
-		klog.Infof("Queueing namespace %s|%s because location %s|%s changed", logicalcluster.From(ns), ns.Name, clusterName, name)
+	for _, obj := range placements {
+		placement := obj.(*schedulingv1alpha1.Placement)
+		klog.Infof("Queueing placement %s|%s because Namespace %q changed", clusterName.String(), placement.Name, ns)
```

Comment:
```suggestion
		klog.Infof("Queueing placement %s|%s because Namespace %q changed", clusterName, placement.Name, ns)
```
no need for `String()`